src/api/Controllers/ExportController.cs

```diff
@@ -0,0 +1,163 @@
+            .ToList();
+
+        foreach (var header in customIdHeaders)
+        {
+            csvWriter.WriteField(header.En.Replace(" ", "", StringComparison.OrdinalIgnoreCase));
+        }
+
+        // Move to the next line
+        await csvWriter.NextRecordAsync().ConfigureAwait(false);
+
+        // Write data for standard fields
+        foreach (var b in boreholes)
+        {
+            var boreholeGeometry = await context.BoreholeGeometry
+                .AsNoTracking()
+                .Where(g => g.BoreholeId == b.Id)
+                .ToListAsync()
+                .ConfigureAwait(false);
+
+            b.TotalDepthTvd = boreholeGeometry.GetTVDIfGeometryExists(b.TotalDepth);
+            b.TopBedrockFreshTvd = boreholeGeometry.GetTVDIfGeometryExists(b.TopBedrockFreshMd);
+            b.TopBedrockWeatheredTvd = boreholeGeometry.GetTVDIfGeometryExists(b.TopBedrockWeatheredMd);
+
+            csvWriter.WriteField(b.Id);
+            csvWriter.WriteField(b.OriginalName);
+            csvWriter.WriteField(b.ProjectName);
+            csvWriter.WriteField(b.Name);
+            csvWriter.WriteField(b.RestrictionId);
+            csvWriter.WriteField(b.RestrictionUntil);
+            csvWriter.WriteField(b.NationalInterest);
+            csvWriter.WriteField(b.LocationX);
+            csvWriter.WriteField(b.LocationY);
+            csvWriter.WriteField(b.LocationXLV03);
+            csvWriter.WriteField(b.LocationYLV03);
+            csvWriter.WriteField(b.LocationPrecisionId);
+            csvWriter.WriteField(b.ElevationZ);
+            csvWriter.WriteField(b.ElevationPrecisionId);
+            csvWriter.WriteField(b.ReferenceElevation);
+            csvWriter.WriteField(b.ReferenceElevationTypeId);
+            csvWriter.WriteField(b.ReferenceElevationPrecisionId);
+            csvWriter.WriteField(b.HrsId);
+            csvWriter.WriteField(b.TypeId);
+            csvWriter.WriteField(b.PurposeId);
+            csvWriter.WriteField(b.StatusId);
+            csvWriter.WriteField(b.Remarks);
+            csvWriter.WriteField(b.TotalDepth);
+            csvWriter.WriteField(b.DepthPrecisionId);
+            csvWriter.WriteField(b.TopBedrockFreshMd);
+            csvWriter.WriteField(b.TopBedrockWeatheredMd);
+            csvWriter.WriteField(b.TotalDepthTvd);
+            csvWriter.WriteField(b.TopBedrockFreshTvd);
+            csvWriter.WriteField(b.TopBedrockWeatheredTvd);
+            csvWriter.WriteField(b.HasGroundwater);
+            csvWriter.WriteField(b.LithologyTopBedrockId);
+            csvWriter.WriteField(b.ChronostratigraphyTopBedrockId);
+            csvWriter.WriteField(b.LithostratigraphyTopBedrockId);
+
+            // Write dynamic fields for custom Ids
+            foreach (var header in customIdHeaders)
+            {
+                var codelistValue = (b.BoreholeCodelists ?? Enumerable.Empty<BoreholeCodelist>()).FirstOrDefault(bc => bc.CodelistId == header.CodelistId)?.Value;
```

Comment:
REQ: Das Konstrukt `b.BoreholeCodelists ?? Enumerable.Empty<BoreholeCodelist>()`  kommt bereits auf Zeile 88 weiter oben vor. Vorschlag: In Variable auslagern.